eslint.config.js

```diff
@@ -0,0 +1,83 @@
+import hypothesis from 'eslint-config-hypothesis';
+import globals from 'globals';
+import tseslint from 'typescript-eslint';
+
+export default tseslint.config(
+  {
+    ignores: [
+      '.yalc/**/*',
+      '.yarn/**/*',
+      'build/**/*',
+      'dist/**/*',
+      '**/vendor/**/*.js',
+      '**/coverage/**/*',
+      'docs/_build/*',
+      // TODO - Lint these files
+      'rollup*.config.js',
+      'src/chrome/content',
+    ],
+  },
+  ...hypothesis,
+  ...tseslint.configs.recommended,
+  {
+    rules: {
+      'prefer-arrow-callback': ['error', { allowNamedFunctions: true }],
+      'object-shorthand': ['error', 'properties'],
+
+      // Upgrade TS rules from warning to error.
+      '@typescript-eslint/no-unused-vars': 'error',
+
+      // Disable TS rules that we dislike.
+      '@typescript-eslint/ban-ts-comment': 'off',
+      '@typescript-eslint/no-empty-function': 'off',
+      '@typescript-eslint/no-explicit-any': 'off',
+      '@typescript-eslint/no-non-null-assertion': 'off',
+      '@typescript-eslint/no-this-alias': 'off',
+
+      // Enforce consistency in cases where TypeScript supports old and new
+      // syntaxes for the same thing.
+      //
+      // - Require `<var> as <type>` for casts
+      // - Require `import type` for type imports. The corresponding rule for
+      //   exports is not enabled yet because that requires setting up type-aware
+      //   linting.
+      '@typescript-eslint/consistent-type-assertions': 'error',
+      '@typescript-eslint/consistent-type-imports': 'error',
+    },
+    languageOptions: {
+      globals: {
+        chrome: false,
+      },
+    },
+  },
+
+  // Scripts
+  {
+    files: [
+      'tests/karma.config.cjs',
+      'tools/*.js',
+      // Entry points which currently get loaded as non-module scripts.
+      'src/unload-client.js',
```

Comment:
`src/unload-client.js` is unlike the other scripts here because it doesn't run in Node. It turns in the browser in a web page. The main difference from other source files is that it is a standalone non-module script due to limitations of Chrome APIs.

We have these categories of files for linting:
- Source files compiled as ESM modules into a bundle
- Classic scripts that run in the browser
- CommonJS scripts that run in Node
- Module scripts that run in Node

Something like this:

```diff
diff --git a/eslint.config.js b/eslint.config.js
index 92250c81..39230686 100644
--- a/eslint.config.js
+++ b/eslint.config.js
@@ -14,7 +14,6 @@ export default tseslint.config(
       'docs/_build/*',
       // TODO - Lint these files
       'rollup*.config.js',
-      'src/chrome/content',
     ],
   },
   ...hypothesis,
@@ -51,16 +50,25 @@ export default tseslint.config(
     },
   },
 
-  // Scripts
+  {
+    // Entry points which get loaded as non-module scripts.
+    files: ['src/unload-client.js'],
+    rules: {
+      strict: ['error', 'global'],
+    },
+    languageOptions: {
+      parserOptions: {
+        sourceType: 'script',
+      },
+    },
+  },
+
+  // CommonJS scripts which run in Node
   {
     files: [
       'tests/karma.config.cjs',
-      'tools/*.js',
-      // Entry points which currently get loaded as non-module scripts.
-      'src/unload-client.js',
     ],
     rules: {
-      'no-console': 'off',
       strict: ['error', 'global'],
     },
     languageOptions: {
@@ -73,11 +81,16 @@ export default tseslint.config(
     },
   },
 
-  // Tools
+  // ESM scripts which run in Node
   {
     files: ['tools/*.js'],
     rules: {
-      strict: 'off',
+      'no-console': 'off',
+    },
+    languageOptions: {
+      globals: {
+        ...globals.node,
+      },
     },
   },
 );
```

Reply:
Yep, makes sense